server/api/src/rsFileChksum.cpp

```diff
@@ -334,6 +335,31 @@ int file_checksum(RsComm* _comm,
     rodsLog(LOG_DEBUG, "file_checksum :: final_scheme [%s]  chkstr_scheme [%s]  hash_policy [%s]",
             final_scheme.data(), chkstr_scheme.c_str(), hash_policy.data());
 
+    irods::file_object_ptr file_ptr{
+        new irods::file_object{_comm, _logical_path, _filename, _resource_hierarchy, -1, 0, O_RDONLY}};
+
+    // Ask resource server if it can read the checksum directly, any error would go
+    // through the normal file read
+    irods::plugin_ptr ptr;
+    irods::resource_ptr resc;
+    irods::error ret_err = file_ptr->resolve(irods::RESOURCE_INTERFACE, ptr);
+    resc = boost::dynamic_pointer_cast<irods::resource>(ptr);
+    if (ret_err.ok()) {
+        std::string checksum_from_resource;
+        std::string final_scheme_str(final_scheme);
+        ret_err = resc->call<const std::string*, std::string*>(
+            _comm, irods::RESOURCE_OP_READ_CHECKSUM_FROM_STORAGE_DEVICE, file_ptr, &final_scheme_str, &checksum_from_resource);
+        if (ret_err.ok()) {
+            strncpy(_calculated_checksum, checksum_from_resource.c_str(), NAME_LEN);
```

Comment:
Add `<cstring>` include and
```suggestion
            std::strncpy(_calculated_checksum, checksum_from_resource.c_str(), NAME_LEN);
```

Reply:
done